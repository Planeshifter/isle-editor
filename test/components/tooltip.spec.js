// MODULES //

import React from 'react';
import renderer from 'react-test-renderer';
import Tooltip from '@isle-project/components/tooltip/main.js';


// TESTS //

it( 'renders a tooltip', () => {
	const tooltip = renderer.create(
		<Tooltip tooltip="I am a tooltip!" >
			<div></div>
		</Tooltip>
	).toJSON();
	expect( tooltip ).toMatchSnapshot();
});

it( 'renders a tooltip with custom placement', () => {
	const tooltip = renderer.create(
		<Tooltip tooltip="I am a tooltip!" placement="right" >
			<div></div>
		</Tooltip>
	).toJSON();
	expect( tooltip ).toMatchSnapshot();
});

it( 'renders an inactive tooltip', () => {
	const tooltip = renderer.create(
		<Tooltip tooltip="I am a tooltip!" show={false} >
			<div></div>
		</Tooltip>
	).toJSON();
	expect( tooltip ).toMatchSnapshot();
});
