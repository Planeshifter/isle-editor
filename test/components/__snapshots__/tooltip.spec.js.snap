// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tooltip /> renders a tooltip 1`] = `
<div
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
/>
`;

exports[`<Tooltip /> renders a tooltip with custom placement 1`] = `
<div
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
/>
`;

exports[`<Tooltip /> renders an inactive tooltip 1`] = `<div />`;
